src/components/LogInModal/LogInModal.js

```diff
@@ -0,0 +1,84 @@
+import { Link } from "react-router-dom/cjs/react-router-dom";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+import { useState } from "react";
+import { useForm } from "../../hooks/useForm";
+import { useFormAndValidation } from "../../hooks/useFormAndValidation";
+import { useHistory } from "react-router-dom/cjs/react-router-dom";
+
+const LogInModal = ({
+  handleCloseModal,
+  isOpen,
+  onSubmit,
+  buttonText,
+  openRegisterModal,
+  inputError,
+}) => {
+  const { values, handleChange, errors, isValid, setErrors } = useFormAndValidation({ email: "", password: "" });
+
+  const history = useHistory();
+
+  const handleFormSubmitLogIn = (e) => {
+    e.preventDefault();
+    if (isValid) {
+      onSubmit(values);
+    }
+  };
+
+  const handleOpenRegisterModal = (e) => {
+    e.preventDefault();
+    openRegisterModal();
+  };
+
+  return (
+    <div className="login">
+      <ModalWithForm
+        title="Log In"
```

Comment:
* Please, pass `isValid` into the props of `ModalWithForm` to disable the submit button if the form is **not** valid